### PR TITLE
Makes Cache an interface to allow custom implementation and mocking

### PR DESCRIPTION
**Pull request by hectorj**

### CODE REVIEW

**Comment by lox** [comment]
Nice, mind rebasing and I'll get this merged once it's tests pass?

**Comment by hectorj** [comment]
Sure, done :)

File: cache.go [hunk 1]
```diff
@@ -28,29 +28,39 @@ const (
...
+	Freshen(res *Resource, keys ...string) error
+}
+
+// cache provides a storage mechanism for cached Resources
+type cache struct {
 	fs    vfs.VFS
 	stale map[string]time.Time
 }
 
+var _ Cache = (*cache)(nil)
```
Comment (lox) [inline cache.go]: What does this do?

Comment (shurcooL) [inline cache.go]: It's a compile time check that the type `*cache` implements the `Cache` interface.

Comment (lox) [inline cache.go]: Ah, clever :)

**Pull request merged**